kvserver: flake in TestGetChecksumNotSuccessfulExitConditions

Found in this CI run https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/6513204?showRootCauses=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&expandBuildTestsSection=true

```
------- Stdout: -------
=== RUN   TestGetChecksumNotSuccessfulExitConditions
I220919 04:00:20.149019 25471 3@pebble/event.go:649  [pebble,s?] 2  [JOB 1] MANIFEST created 000001
I220919 04:00:20.149172 25471 3@pebble/event.go:677  [pebble,s?] 3  [JOB 1] WAL created 000002
I220919 04:00:20.149313 25471 3@pebble/event.go:645  [pebble,s?] 4  upgraded to format version: 002
I220919 04:00:20.149338 25471 3@pebble/event.go:645  [pebble,s?] 5  upgraded to format version: 003
I220919 04:00:20.149353 25471 3@pebble/event.go:645  [pebble,s?] 6  upgraded to format version: 004
I220919 04:00:20.149390 25471 3@pebble/event.go:645  [pebble,s?] 7  upgraded to format version: 005
I220919 04:00:20.149413 25471 3@pebble/event.go:645  [pebble,s?] 8  upgraded to format version: 006
I220919 04:00:20.149431 25471 3@pebble/event.go:645  [pebble,s?] 9  upgraded to format version: 007
I220919 04:00:20.149451 25471 3@pebble/event.go:645  [pebble,s?] 10  upgraded to format version: 008
I220919 04:00:20.149463 25471 3@pebble/event.go:645  [pebble,s?] 11  upgraded to format version: 009
I220919 04:00:20.149476 25471 3@pebble/event.go:645  [pebble,s?] 12  upgraded to format version: 010
I220919 04:00:20.150044 25497 3@pebble/event.go:669  [pebble,s?] 13  [JOB 4] all initial table stats loaded
    replica_consistency_test.go:128:
          Error Trace:  /go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_consistency_test.go:128
          Error:        Target error should be in err chain:
                        expected: "context deadline exceeded"
                        in chain: "checksum computation did not start in time for (ID = 6ce8d64c-8f24-4cb8-85d3-f516c241e895)"
                          "checksum computation did not start in time for (ID = 6ce8d64c-8f24-4cb8-85d3-f516c241e895)"
          Test:         TestGetChecksumNotSuccessfulExitConditions
--- FAIL: TestGetChecksumNotSuccessfulExitConditions (1.04s)
```

cc @shralex for triage

Jira issue: CRDB-19706

--- a/pkg/kv/kvserver/replica_consistency_test.go
+++ b/pkg/kv/kvserver/replica_consistency_test.go
@@ -124,8 +124,10 @@ func TestGetChecksumNotSuccessfulExitConditions(t *testing.T) {
 
 	// Context is canceled during the initial waiting.
 	id = uuid.FastMakeV4()
+	ctx, cancel = context.WithCancel(context.Background())
+	cancel()
 	rc, err = tc.repl.getChecksum(ctx, id)
-	require.ErrorIs(t, err, context.DeadlineExceeded)
+	require.ErrorIs(t, err, context.Canceled)
 	require.Nil(t, rc.Checksum)
 	requireChecksumTaskNotStarted(id)
 }

